Add wxPython and use install_requires

smartsammler opened the pull request:
* Add wxPython as a depenency
* Use `install_requires` so dependencies are installed if one installs this package
  See e. g. https://packaging.python.org/discussions/install-requires-vs-requirements/

smartsammler committed:
Add wxPython and use install_requires

* Add wxPython as a depenency
* Use `install_requires` so dependencies are installed if one installs this package

zhangt58 commented on setup.py:
```diff
@@ -29,5 +29,6 @@ def readme():
         url = "http://archman.github.io/beamline/",
         scripts  = [os.path.join('scripts',sn) for sn in scriptnames],
         requires = requiredpackages,
```
Then this line should be removed.

smartsammler replied:
Ok, I did not know the keyword hence I left it in. I will change that.

zhangt58 commented on setup.py:
```diff
@@ -14,7 +14,7 @@ def readme():
         return f.read()
 
-requiredpackages = ['pyrpn', 'h5py', 'numpy', 'matplotlib', 'pyepics']
+requiredpackages = ['pyrpn', 'h5py', 'numpy', 'matplotlib', 'pyepics', 'wxPython']
```
The only concern about adding `wxPython` as one of the deps is the compatibility since pip can only install 4.x version, which is not working for the present codebase.

smartsammler replied:
Good point. Probably it is better to mention the dependency on wxPython in the README and leave it out here.

smartsammler committed:
Delete `requires` and `wxPython` dependency

`wxPython` is not easily installable via pip on all Linux distributions without explicitly providing the path to the wheel, yet.

`requires` is obsolete, because of the use of `install_requires`

smartsammler committed:
Add note about the dependencies

`wxPython` is not installed as a dependency and some people tend to skip the installation chapter of documentation the first time using it.
